orttraining/orttraining/training_ops/cpu/nn/layer_norm.cc

```diff
@@ -58,8 +58,10 @@ Status LayerNormGrad<T, simplified>::Compute(OpKernelContext* op_kernel_context)
   const Tensor* X = op_kernel_context->Input<Tensor>(input_index++);
   const auto& X_shape = X->Shape();
   const auto axis = HandleNegativeAxis(axis_, X_shape.NumDimensions());
-  const auto N = X_shape.SizeToDimension(axis);
-  const auto M = X_shape.SizeFromDimension(axis);
+  ORT_ENFORCE(X_shape.SizeToDimension(axis) <= std::numeric_limits<int>::max());
```

Comment:
I'm not 100% sure if this will work, but could we use `Eigen::Index` instead of `int`? (I think this is the type that is used when constructing various forms of `Eigen::Map` below.

https://eigen.tuxfamily.org/dox/namespaceEigen.html#a62e77e0933482dafde8fe197d9a2cfde

Reply:
Eigen::Index is an alias of ptrdiff_t, both should work.

Reply:
Eigen::Index seems to work. Well it builds, I assume any applicable unit tests will be run by the ci pipeline?